src/createServer.js

```diff
@@ -1,8 +1,44 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const url = require('url');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'text/plain');
+
+    const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+
+    if (!normalizedUrl.pathname.startsWith('/file')) {
+      res.statusCode = 400;
+      res.statusMessage = 'Path should start with "/file/"';
+      res.end();
+    } else {
+      const requestedPath =
+        normalizedUrl.pathname === '/file' ||
+        normalizedUrl.pathname === '/file/'
+          ? 'index.html'
+          : normalizedUrl.pathname.slice(6);
+
+      if (requestedPath.includes('//')) {
+        res.statusCode = 404;
+        res.statusMessage = 'File path should not include double-slash';
+        res.end('File path should not include double-slash');
+      }
+
+      fs.readFile(`./public/${requestedPath}`, (err, data) => {
+        if (!err) {
+          res.statusCode = 200;
+          res.end(data);
+        } else {
+          res.statusCode = 404;
+          res.statusMessage = 'File is not existing';
```

Comment:
The status message 'File is not existing' is grammatically incorrect. Consider changing it to 'File does not exist'.